src/components/Interpretations/common/getInterpretationAccess.js

```diff
@@ -1,9 +1,23 @@
-const isCreatorOrSuperuser = (object, currentUser) =>
-    object?.createdBy.id === currentUser?.id ||
-    currentUser?.authorities.has('ALL')
+// For backwards compatibility
+// accept both Set (from the old d2.currentUser object) and array
+const hasAuthority = (authorities = [], authority) => {
+    if (Array.isArray(authorities)) {
+        return authorities.includes(authority)
+    }
+    return typeof authorities.has === 'function'
+        ? authorities.has(authority)
+        : false
+}
```

Comment:
Actually, taking my own advice, we should probably not have default values either:

```suggestion
const hasAuthority = (authorities, authority) => {
    if (!authority || typeof authority !== 'string) {
        throw new Error(`"hasAuthority" requires "authority" to be a populated string but received ${authority}`)
    }
    else if (Array.isArray(authorities)) {
        return authorities.includes(authority)
    } else if (typeof authorities.has === 'function') {
        return authorities.has(authority)
    } else {
        throw new Error(`"hasAuthority" requires "authorities" to be an array or set of authorities (strings)`)
    }
}
```